src/constraints/constraint_min_scheduled_outage_duration.jl

```diff
@@ -0,0 +1,63 @@
+#############################################################################
+# Copyright (C) 2017 - 2023  Spine Project
+#
+# This file is part of SpineOpt.
+#
+# SpineOpt is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# SpineOpt is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+@doc raw"""
+The unit must be taken out of service for maintenance for a duration equal to scheduled_outage_duration:
+
+```math
+\sum_{t} v^{units\_out\_of\_service}_{(u,s,t)}duration_t \geq scheduled\_outage\_duration_{(u,s,t)}number\_of\_units_u \quad \forall u \in unit, \, \forall (s,t)
+```
+
+"""
+function add_constraint_min_scheduled_outage_duration!(m::Model)
+    @fetch units_out_of_service = m.ext[:spineopt].variables
+    t0 = _analysis_time(m)
+    m.ext[:spineopt].constraints[:min_scheduled_outage_duration] = Dict(
+        (unit=u, stochastic_path=s_long, t=t_long) => @constraint(
+            m, 
+            + sum(
+                units_out_of_service[u, s, t] * duration(t)
+                for (u, s, t) in units_on_indices(m; unit=u);
+                init=0
+            ) 
+            >= ( + scheduled_outage_duration[(unit=u, stochastic_scenario=s_long, analysis_time=t0, t=t_long)]
```

Comment:
Maybe we need to break the line after the left parenthesis?

Reply:
Fixed this in [latest commit](https://github.com/spine-tools/SpineOpt.jl/commit/21097e41186dd36e83125e591bcfd34989910eeb)